ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -3584,39 +3584,50 @@ IHqlExpression * HqlCppTranslator::getRtlFieldKey(IHqlExpression * expr, IHqlExp
                  Theoretically with an inline record definition for a field it might be possible to make an ifblock dependent on something other than the most
                  immediate parent record, but it would be extremely pathological, and probably wouldn't work in lots of other ways.
     */
-
+    isPayload = false;
     bool contextDependent = false;
     LinkedHqlExpr extra = rowRecord;
     switch  (expr->getOperator())
     {
     case no_field:
-        switch (expr->queryType()->getTypeCode())
+    {
+        ColumnToOffsetMap * map = queryRecordOffsetMap(rowRecord, false);
+        AColumnInfo * root = map->queryRootColumn();
+        AColumnInfo * columnInfo = root->lookupColumn(expr);
+
+        if (columnInfo->isPayloadField())
         {
-        case type_bitfield:
+            isPayload = true;
+            contextDependent = true;
+        }
+        else
```

Comment:
not sure this should be else.  bitfields in payloads still need the code that follows